src/compile/model.ts

```diff
@@ -41,24 +41,39 @@ export interface Component {
   mark: VgMarkGroup[];
 }
 
-class NameMap {
+export class NameMap {
   private _nameMap: Dict<string>;
 
   constructor() {
     this._nameMap = {} as Dict<string>;
   }
 
   public rename(oldName: string, newName: string) {
+    if (this._nameMap[newName] === oldName) {
+      console.error('Cannot rename ' + oldName + ' to ' + newName);
```

Comment:
Do we really need this clause?  

In general, `console.error` doesn't really help us because we don't always see it.